feat(rdb): monitor metrics MTA-4706 (#3440)
scaleway/docs-content: managed-databases/postgresql-and-mysql/how-to/monitor-databases-cockpit.mdx

2. From the Scaleway console [Cockpit overview page](https://console.scaleway.com/cockpit/overview), click **Open dashboards** in the top right corner.

3. Enter your Grafana username and password to log in.

    You are directed to the Scaleway Cockpit dashboards homepage.

## How to view and understand your Managed Database metrics

1. Click **Open dashboards** on the top right from the Scaleway Cockpit dashboards homepage.
2. Click either **RDB MySQL Overview** or **RDB PostgreSQL Overview**, depending on your engine of choice, on the left side menu.

    You are directed to the RDB Overview dashboard.

    <Lightbox src="scaleway-grafana-rdb.webp" alt="" />

Metrics are refreshed every minute, and report on the following data:

- **CPU usage**
- **Memory usage**
- **Total connections**
- **Disk usage** (in % and in Bytes)
- **Replication lag in seconds** (when High Availability mode is activated or when Read Replicas are created).

### Choosing metric parameters

At the top of the dashboard, you can configure the following parameters, that control which metrics are displayed:

- **Resource name** - Select a specific Database Instance to monitor.

> - **Node** - Choose a specific node (main, replica, or Read Replica). In an High Availability (HA) cluster, failovers may show the same node as both main and replica. For nodes in an HA cluster, you might see two verions of the same node, the main and the replica. The replica will start appearing as if any failovers happen and will show only metrics for any activity that happened over the failover period.

Reviewer: ```suggestion
- **Node** - Choose a specific node (main, replica, or Read Replica). In a High Availability (HA) cluster, failovers may show the same node as both main and replica. For nodes in an HA cluster, you might see two versions of the same node, the main and the replica. The replica will start appearing as if any failovers happen and will show only metrics for any activity that happened over the failover period.
```

Reviewer 2: This part is not very clear to me. I have another suggestion:
"In an High Availability (HA) cluster, both nodes might switch from main to replica and vice versa. So you might see two versions of the same node, as main and as replica. If you select main (respectively replica) for a specific node, you will only see activity of the node when he was main (resp. replica)"

wdyt?

Author: ```suggestion
- **Node** - Choose a specific node (main, replica, or Read Replica). In a High Availability (HA) cluster, both nodes might switch from main to replica and vice versa. So you might see two versions of the same node, as main and as replica. If you select main for a specific node, you will only see the node's activity when it was the main, and vice-versa.
```

- **Timeframe** - Get metrics from the last five minutes up to 31 days (free plan) and 365 days (premium plan).

### Viewing logs

The log section is displayed below the metrics section.

Click **Logs** to collapse the tab and view your logs.

Log configuration is done by Scaleway. Note that:

- The `log_error_verbosity` parameter is set to `VERBOSE` for PostgreSQL and `2` for MySQL by default.
- On PostgreSQL, all queries longer than 5 seconds are logged. This parameter cannot be edited.
- On MySQL, slow queries are not activated by default, but can be activated in the [advanced settings](/managed-databases/postgresql-and-mysql/how-to/configure-advanced-settings) of the Database Instance. The setting is called `slow_query_log` and the time can be edited with `long_query_time`.

> 

Reviewer: ```suggestion
```

## How to activate managed alerts

You can activate managed alerts for your Managed Databases with Scaleway Cockpit.

<Message type="note">
  Refer to the [How to activate managed alerts](/observability/cockpit/how-to/activate-managed-alerts/) to learn how to do so.
</Message>

By default, three managed alerts are configured for Managed Databases:

- `MySQLHighStorageUsage` or `PostgreSQLHighStorageUsage`- You are alerted if disk usage is higher than 80% for 10 consecutive minutes.
- `MySQLHighCPULoad` or `PostgreSQLHighCPULoad`- You are alerted if CPU usage is higher than 80% for 10 consecutive minutes.
- `MySQLTooManyConnections` or `PostgreSQLTooManyConnections`- You are alerted if your connection usage is higher than 80% for 10 consecutive minutes.

> You can also configure [your own managed alerts](/observability/cockpit/how-to/configure-alerts-for-scw-resources/), and configure values for each of the parameters below to define when you want to be alerted:

Reviewer: ```suggestion
You can also configure [your own managed alerts](/observability/cockpit/how-to/configure-alerts-for-scw-resources/) and set values for each of the parameters below to define when you want to be alerted:
```

### MySQL

| Alert | Description |
| ------------ | ------------------- |
| `rdb_instance_mysql_mysql_global_status_threads_connected` | Total active connections on MySQL |
| `rdb_instance_mysql_mysql_global_variables_max_connections` | Maximum number of connections configured |
| `rdb_instance_mysql_mysql_slave_status_seconds_behind_master` | Replica lag in seconds |
| `rdb_instance_mysql_node_cpu_seconds_total` | Total amount of CPU time consumed by the Database Instance, measured in seconds |
| `rdb_instance_mysql_node_filesystem_avail_bytes` | Total amount of remaining available space, in bytes |
| `rdb_instance_mysql_node_filesystem_size_bytes` | Total size of the filesystem, in bytes |
| `b_instance_mysql_node_memory_MemAvailable_bytes` | Amount of memory available |
| `rdb_instance_mysql_node_memory_MemTotal_bytes` | Total installed physical memory (RAM) |

> 

Reviewer: ```suggestion
```

### PostgreSQL

| Alert | Description |
| ------------ | ------------------- |
| `rdb_instance_postgresql_node_cpu_seconds_total` | Total amount of CPU time consumed by the Database Instance, measured in seconds |
| `db_instance_postgresql_node_filesystem_avail_byte` | Total amount of remaining available space, in bytes |
| `rdb_instance_postgresql_node_filesystem_size_bytes` | Total size of the filesystem, in bytes |
| `db_instance_postgresql_node_memory_MemAvailable_bytes` | Amount of memory available |
| `rdb_instance_postgresql_node_memory_MemTotal_bytes` | Total installed physical memory (RAM) |
| `rdb_instance_postgresql_pg_replication_lag` | Replica lag in seconds |
| `rdb_instance_postgresql_pg_settings_max_connection` | Maximum number of concurrent Database Instance connections  |
| `rdb_instance_postgresql_pg_stat_database_numbackends` | Number of active Database Instance connections. |

> 

Reviewer: ```suggestion
```
